### PR TITLE
Slight cleanup in SpirvLowerGlobal

### PR DESCRIPTION
**Pull request by Flakebi**
This uses one bitcast less than the for loop and takes on less line.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit b70f76da47e6eedf82b8744d2b8ebc757ac1f167
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 32d42f58e8f4c3de99490c36d5a2091f8c59a673</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 63761ecfd2d3cfa61a756ce0a72e1ca1cdae333a</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev fde3d4e2bd90743606ca7b724bf62a53cacf4615</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/144501)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26261/48174 (54.5%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21913/48174 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26320/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21854/48174 (45.4%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>

File: llpc/lower/llpcSpirvLowerGlobal.cpp [hunk 1]
```diff
@@ -1666,23 +1666,22 @@ void SpirvLowerGlobal::lowerBufferBlock() {
           if (!isa<Instruction>(user))
             continue;
 
-          GetElementPtrInst *getElemPtr = dyn_cast<GetElementPtrInst>(user);
+          // Skip all bitcasts, looking for a GEP.
+          Value *inst = user;
```
Comment (kuhar) [inline llpc/lower/llpcSpirvLowerGlobal.cpp:1670]: Could we use `stripPointerCasts`? If all-zero GEPs would be the issue, maybe we can create a free function, something like `Value* stripBitcasts(Value *val, SmallVectorImpl<BitCastInst>& stripped)`

File: llpc/lower/llpcSpirvLowerGlobal.cpp [hunk 2]
```diff
@@ -1666,23 +1666,22 @@ void SpirvLowerGlobal::lowerBufferBlock() {
...
-              BitCastInst *const bitCast = dyn_cast<BitCastInst>(user);
-              assert(bitCast);
-
-              bitCastsToModify.push_back(bitCast);
-              continue;
-            }
+            BitCastInst *const bitCast = dyn_cast<BitCastInst>(user);
+            assert(bitCast);
+
+            bitCastsToModify.push_back(bitCast);
```
Comment (kuhar) [inline llpc/lower/llpcSpirvLowerGlobal.cpp:1683]: Why do we modify only the first bitcast?

**Comment by Flakebi** [comment]
Always these thought-provoking reviews… ;)
Here we go with less lines, the loop didn’t really make sense.

**Comment by amdvlk-admin** [comment]
### Test summary for commit e3ef0fa145f6e164ffaffc0965a97bbf6095f133
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 32d42f58e8f4c3de99490c36d5a2091f8c59a673</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 63761ecfd2d3cfa61a756ce0a72e1ca1cdae333a</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev fde3d4e2bd90743606ca7b724bf62a53cacf4615</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/96327)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26321/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21853/48174 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>

**Review by kuhar — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**